MIND - Blog (#2023)
FlowFuse/website: src/blog/2024/05/node-red-mind-stack-with-flowfuse.md

## Optimizing IoT with the MIND Stack

It often makes sense to deploy a full MING stack, but in some deployments, it might be necessary to deploy a more simplistic version of the MING stack.  This is where MIND comes into play.

### FlowFuse MQTT: Simplified Communications

FlowFuse has a built-in feature called projects, which leverages MQTT, that allows the communication of data between FlowFuse runtimes of Node-RED.  One caveat is that this MQTT broker is only available within the FlowFuse platform. What this means is there needs to be some form of translation to be done within Node-RED.  This isn’t a big deal for small deployments, because often Node-RED runtimes at the edger are collecting data from various sources that aren’t MQTT.  The flow of data is as follows: 

> Sensor > Node-RED(FlowFuse Device Agent) > MQTT Encapsulated by FlowFuse > Node-RED(FlowFuse Platform) > InfluxDB

Author: @Yndira-E Could you make an artwork for this data flow?  I will make a ticket if it sounds reasonable.

Reviewer: Hi @gdziuba sure, open an art request and I'll take care of it.

It would be great if you could provide some examples or even a description of what image, icon, or logo would represent each state of the flow. I'm not sure how "MQTT Encapsulated by FlowFuse" should look like, or the sensor. Is everything but the sensor and InfluxDB meant to be represented "inside" FlowFuse?

Author: https://github.com/FlowFuse/website/issues/2029

Reviewer: I've just pushed it and added it in the .md file, please check it's ok.

---
title: The MIND stack with Node-RED and FlowFuse Dashboard 2.0
subtitle: Leveraging FlowFuse for a Lighter MIND Stack (MIND = MQTT, InfluxDB, Node-RED, Dashboard 2.0)
description: Our objective is to streamline the deployment of the MIND stack, enhancing its usability without compromising functionality.

> date: 2024-05-07

Author: ```suggestion
date: 2024-05-14
```

<!--more-->

## FlowFuse: Integrated IoT Deployment

With FlowFuse we have bundled up 3 of those 4.  First and foremost, the FlowFuse platform deploys Node-RED in two main ways.  A cloud based deployment and remote deployment with management from the FlowFuse platform.  Part of this management piece leverage MQTT between the devices for management.  Because of this we have a built in MQTT broker that allows for communication between Node-RED instance within the FlowFuse platform.  Lastly, with the release of Dashboards 2.0 from FlowFuse we have introduced the next generation of visualizations for the Node-RED platform.

## Optimizing IoT with the MIND Stack

It often makes sense to deploy a full MING stack, but in some deployments, it might be necessary to deploy a more simplistic version of the MING stack.  This is where MIND comes into play.

### FlowFuse MQTT: Simplified Communications

> FlowFuse has a built-in feature called projects, which leverages MQTT, that allows the communication of data between FlowFuse runtimes of Node-RED.  One caveat is that this MQTT broker is only available within the FlowFuse platform. What this means is there needs to be some form of translation to be done within Node-RED.  This isn’t a big deal for small deployments, because often Node-RED runtimes at the edger are collecting data from various sources that aren’t MQTT.  The flow of data is as follows: 

Reviewer 2: ```suggestion
FlowFuse has a built-in feature called project link nodes, which leverages MQTT, that allows the communication of data between FlowFuse runtimes of Node-RED. One caveat is that this MQTT broker is only available within the FlowFuse platform. What this means is there needs to be some form of translation to be done within Node-RED.  This isn’t a big deal for small deployments, because often Node-RED runtimes at the edger are collecting data from various sources that aren’t MQTT. The flow of data is as follows: 
```

Sensor > Node-RED(FlowFuse Device Agent) > MQTT Encapsulated by FlowFuse > Node-RED(FlowFuse Platform) > InfluxDB

!["Screenshot showing the flow of data: Sensor > Node-RED(FlowFuse Device Agent) > MQTT Encapsulated by FlowFuse > Node-RED(FlowFuse Platform) > InfluxDB"](./images/sensor-data-mqtt-node-red-dashboard-influxdb.png ""){data-zoomable}

> A key benefit of having the MQTT broker encapsulated by FlowFuse is that it becomes extremely easy to set up.  With the project nodes, there isn’t a need to configure certificates or security credentials because it is only exposed internally to the FlowFuse platform.

Reviewer 2: ```suggestion
A key benefit of having the MQTT broker encapsulated by FlowFuse is that it becomes extremely easy to set up. With the project nodes, there isn’t a need to configure certificates or security credentials because it is only exposed internally to the FlowFuse platform. Furthermore, the access and control limits are applied by the platform so your data remains within the same FlowFuse Team.
```

### InfluxDB: Efficient Data Storage

> InfluxDB doesn’t change much in the MIND configuration compared to how it is deployed in the MING stack, minus the fact that the data will be stored next to the FlowFuse Platform instead of the cloud.  

Reviewer 2: > minus the fact that the data will be stored next to the FlowFuse Platform

What does this mean?

Author: updated.

### Data Visualization with Dashboard 2.0

FlowFuse's Dashboard 2.0 makes it incredibly easy to visualize your data.  You can create beautiful and informative dashboards in minutes. Simply drag and drop your desired widgets onto the canvas, and configure them with a few clicks.  

We have gone through the step of providing you with an example dashboard for visualizing your MIND stack.   

> _Sample Dashboard_

Reviewer 2: ?

Author: Updated with renderFlow.  Thanks for catching this.

## Conclusion

While we believe that the MING stack is here to stay, we believe as the market matures, it should offer many different variations suitable for each customer's needs.  However, we find that the MIND offering does fill a niche in the market that may better suit your needs.  None the less, choosing FlowFuse to manage your Node-RED runtimes will ensure that your applications will be secure, scalable, and easily manageable.  Allowing your domain experts to take control and extend their knowledge bringing increased value to industrial facilities around the world.

> Ready to get started, sign up today for your free trial. 

Reviewer 2: Link?

Author: ```suggestion
Ready to get started, sign up today for your [free trial](https://app.flowfuse.com/account/create). 
```
